File: examples/linreg/example_bptest.R
library("mmstat4")
staedtemieten <- ghload("staedtemieten.rds")
x  <- staedtemieten[complete.cases(staedtemieten),]
lm <- lm (Miete~Fläche, data=x)
summary(lm)
plot(x$Fläche, residuals(lm))
abline(h=0, col="red")
#
library("lmtest")
bptest(Miete~Fläche, data=x)

File: examples/linreg/example_bptest.py
import pandas as pd
import statsmodels.api as sm
import statsmodels.formula.api as smf
from statsmodels.stats.diagnostic import (
  het_breuschpagan,
  het_white)
staedtemieten = r["mmstat4::ghload"]("staedtemieten.rds")
x  = staedtemieten.dropna()
lm = smf.ols('Miete ~ Fläche', data=x).fit()
X  = sm.add_constant(x['Fläche'])
# second test: F Test (H0: all coeff. equal 0)
print(het_breuschpagan(lm.resid, X))
print(het_white(lm.resid, X))       
